simmechanics/CMakeLists.txt

```diff
@@ -93,28 +93,96 @@ macro(generate_icub_simmechanics)
     else()
       set(GAZEBO_ASSIGNED_INERTIAS "")
     endif()
+
+    # If we are generating the icub2.5+ model we need to add the xsens IMU, change the mesh folder
+    # and change the rotation axis list that need to be reversed.
+    if(GIVTWO_ICUB_PLUS)
+      set(MESH_FOLDER "filenameformatchangeext: \"package://iCub/meshes/simmechanics/2-5_plus/%s-binary.stl\"")
+      set(ANKLE_PITCH_ROM "-50,20")
+      set(CUSTOM_EPSILON "epsilon: 2e-7")
+      set(XSENS_IMU_FRAME
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    frameReferenceLink: root_link
+    exportedFrameName: root_link_imu_frame
+")
+      set(XSENS_IMU_SENSOR
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    linkName: root_link
+    exportFrameInURDF: No
+    sensorName: root_link_xsens_imu
+    sensorType: \"imu\"
+    updateRate: \"400\"
+    sensorBlobs:
+    - |
+        <plugin name=\"root_link_xsens_imu_plugin\" filename=\"libgazebo_yarp_imu.so\">
+            <yarpConfigurationFile>model://iCub/conf/gazebo_icub_inertial.ini</yarpConfigurationFile>
+        </plugin>
+")
+      set(REVERSE_ROTATION_AXIS
+"reverseRotationAxis:
+  l_shoulder_roll
+  l_elbow
+  l_hip_yaw
+  l_knee
+  l_ankle_pitch
+  l_hip_pitch
+  r_ankle_roll
+  r_shoulder_pitch
+  r_shoulder_yaw
+  r_elbow
+  r_wrist_prosup
+  torso_pitch
+  neck_roll
+")
+    else()
+      set(MESH_FOLDER "filenameformatchangeext: \"package://iCub/meshes/simmechanics/%s-binary.stl\"")
+      set(ANKLE_PITCH_ROM "-35,35")
+      set(CUSTOM_EPSILON "")
+      set(XSENS_IMU_FRAME "")
+      set(XSENS_IMU_SENSOR "")
+      set(REVERSE_ROTATION_AXIS
+"reverseRotationAxis:
+  l_shoulder_roll
+  l_elbow
+  l_hip_yaw
+  l_knee
+  l_ankle_pitch
+  r_hip_pitch
```

Comment:
Why is `r_hip_pitch` inverted instead of `l_hip_pitch`?

Reply:
The axis that are inverted are magically changing whenever we change the Creo model. 😕

Reply:
This comes from the CAD model. In theory, it is possible to control the orientation of the axis but is not practical. It is easier to perform the change in the YAML file. Probably, since the upper body is the same for the iCub2.5 and iCub2.5+ we can change only the leg axis.

Reply:
Ok.